include/aws/common/windows/rw_lock.inl

```diff
@@ -0,0 +1,82 @@
+#ifndef AWS_COMMON_WINDOWS_RW_LOCK_INL
+#define AWS_COMMON_WINDOWS_RW_LOCK_INL
+
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int aws_rw_lock_init(struct aws_rw_lock *lock) {
```

Comment:
what's the point of putting these in cpp guards if we aren't doing dll semantics?
Even then why would we export platform specific symbols?

Reply:
You're right, we probably don't need the cpp guards. Where do you see this exporting platform specific symbols?

Reply:
shouldn't these implementations be marked `inline` to prevent the compiler from borking?